lib/model.js

```diff
@@ -121,14 +192,14 @@ module.exports = class Model {
 			let type = attribute && exists(attribute.type) && attribute.type;
 			let transform = attribute && isFunction(attribute.transform) && attribute.transform;
 
-			let value = values && exists(values[key]) ? values[key] : null;
+			let value = values && exists(values[key]) ? values[key] : undefined;
 			value = ( type !== 'function' && isFunction(value) ) ? value(key, type) : value;
```

Comment:
yes basically if the type is function then it means we should test the value to be a function but if its not a function and it sees value as a function it runs it